### PR TITLE
Update rules dist for new php-cs-fixer version

### PR DESCRIPTION
**Pull request by jesseleite**
## Why?

The TLDR is that php-cs-fixer 3.x has a bunch of [breaking changes](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v3.0.0/UPGRADE-v3.md), and we want to be able to use the latest version.

## What's changed?

We [copied our initial rules file](https://github.com/statamic/cms/commit/68c2df96b49fd47050202eed3b5b66f61b6c1d56) from [Laravel Shift](https://gist.github.com/laravel-shift/cab527923ed2a109dda047b97d53c200). After following the upgrade guide myself, it looks like they did a more thorough job upgrading their rules, so I am once again copying most of this from their gist, then we can tweak to suit us 👍 

- [x] Rename `.php_cs.dist` to `.php-cs-fixer.dist.php`.
- [x] Re-sync with [Laravel Shift's](https://gist.github.com/laravel-shift/cab527923ed2a109dda047b97d53c200) php-cs-fixer rules gist.
- [x] Tweak to our liking.
    - [x] Re-add our `no_unused_imports` rule.
    - [x] Remove enforced line spacing between properties, constants, and trait imports?
        ```php
        'class_attributes_separation' => [
            'elements' => [
                'method' => 'one',
                'const' => 'one',         // seems overkill, remove this?
                'property' => 'one',      // seems overkill, remove this?
                'trait_import' => 'none', // seems overkill, remove this?
            ]
        ]
        ```

## How to test?

- Pull down this branch
- Run `composer global require friendsofphp/php-cs-fixer ^3.0`
- Try running `php-cs-fixer fix src` and look at the unstaged changes
- Help me tweak

### CODE REVIEW

**Comment by FrittenKeeZ** [comment]
Just curious... why don't you extend the `@PSR12` rules and adjust the last ones?
